robocorp-code/src/robocorp_code/inspector/image/_bridge.py

```diff
@@ -0,0 +1,187 @@
+import json
+import os
+import subprocess
+import sys
+import logging
+from pathlib import Path
+from typing import Optional
+from uuid import uuid4
+
+from robocorp_code.inspector.common import (
+    STATE_CLOSED,
+    STATE_INITIALIZING,
+    STATE_PICKING,
+)
+
+DEFAULT_CONFIDENCE = 80
+
+
+def _base64_to_image(img: str):
+    # pylint: disable=import-outside-toplevel
+    from .vendor.utils import base64_to_image
+
+    # making sure we remove the unnecessary composition
+    img = img.replace("data:image/png;base64,", "")
+
+    return base64_to_image(img)
+
+
+class ImageBridge:
+    from robocorp_ls_core.protocols import IEndPoint
+
+    """Javascript API bridge for image template locators."""
+
+    def __init__(self, endpoint: Optional[IEndPoint] = None, logger=None) -> None:
+        self.endpoint = endpoint
+
+        self.logger = (
+            logger if logger is not None else logging.getLogger(self.__class__.__name__)
+        )
+
+        self.snipping_process: Optional[subprocess.Popen] = None
+
+    def _launch_snipper(self):
+        #: Create the appropriate environment and launch the process
+        # - this is blocking and waits for the process to finish
+        environ = os.environ.copy()
+        environ.pop("PYTHONPATH", "")
+        environ.pop("PYTHONHOME", "")
+        environ.pop("VIRTUAL_ENV", "")
+        environ["PYTHONIOENCODING"] = "utf-8"
+        environ["PYTHONUNBUFFERED"] = "1"
+        cmd = [
+            sys.executable,
+            str(Path(__file__).parent / "_snipping_tool.py"),
+        ]
+        if self.endpoint is not None:
+            self.endpoint.notify("$/imageInspectorState", {"state": STATE_PICKING})
+
+        self.snipping_process = subprocess.Popen(
+            cmd,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            stdin=subprocess.PIPE,
+            env=environ,
+            bufsize=0,
+        )
+        stdout, _ = self.snipping_process.communicate()
+
+        # If you want to capture the output, you can use stdout.decode('utf-8') to convert it to a string
+        output = stdout.decode("utf-8")
```

Comment:
Use `stdout.decode("utf-8", "replace")` here (so that it won't fail with a UnicodeDecodeError in case some non utf-8 char is printed).